lib/addons.sh

```diff
@@ -487,6 +487,32 @@ function bashio::addon.options() {
     bashio::addons "${slug}" "addons.${slug}.options" '.options'
 }
 
+# ------------------------------------------------------------------------------
+# Sets option for this add-on.
+#
+# Arguments:
+#   $1 Config Key to Set (required)
+#   $2 Value to set (optional, default:null, if null will remove the key pair)
+#   $3 Add-on slug (optional, default: self)
+# ------------------------------------------------------------------------------
+function bashio::addon.option.set() {
+    local key=${1}
+    local value=${2:-"$null"}
+    local slug=${3:-'self'}
+    bashio::log.trace "${FUNCNAME[0]}" "$@"
+    local options
+    if bashio::var.has_value "${value}"; then
+      options=$(bashio::addons "${slug}" "addons.${slug}.options" '.options' | \
+      jq --arg value "$value" --arg key "$key" '.[$key]=$value')
+    else
+      options=$(bashio::addons "${slug}" "addons.${slug}.options" '.options' | \
+      jq --arg key "$key" 'del(.[$key])')
+    fi
+    bashio::api.supervisor POST "/addons/${slug}/options" \
+        "{\"options\": ${options}}"
+    bashio::cache.flush_all
+}
```

Comment:
I think will work

```suggestion
function bashio::addon.option.set() {
    local key=${1}
    local value=${2:-"$null"}
    local slug=${3:-'self'}
    local options
    local payload

    bashio::log.trace "${FUNCNAME[0]}" "$@"

	options=$(bashio::addons "${slug}" "addons.${slug}.options"

    if bashio::var.has_value "${value}"; then
      options=$(bashio::jq "${options}" ".[${key}]=\"${value}\"")
    else
      options=$(bashio::jq "${options}" "del(.[${key}])")
    fi
    
    payload=$(bashio::var.json options "${options}")    
    bashio::api.supervisor POST "/addons/${slug}/options" "${payload}"

    bashio::cache.flush_all
}
```

Maybe rename it to simply `bashio::addon.option`? (Although that isn't clear as well, but we don't use `set` anywhere yet... 🤷

Reply:
Makes sense, will take another look when I have time (as its been a while).

One challenge I believe was everything was posted as a string, which was an issue for Boolean I believe, or does the var.json fix that?

Reply:
Oh right this will cause issues with non-string values 🤔

Reply:
Was going to look at parsing switches to jq, but then it got more complicated 😉

Reply:
Maybe we need to introduce `bashio::addon.option` (note the non-plural)

Get: `bashio::addon.option <key>`
Set: `bashio::addon.option <key> <value>`

For the value, we can reuse the logic from `bashio::var.json` that has magic to handle other types (`^` as an escape).